core/src/trezor/wire/message_handler.py

```diff
@@ -0,0 +1,234 @@
+from micropython import const
+from typing import TYPE_CHECKING
+
+from storage.cache_common import InvalidSessionError
+from trezor import log, loop, protobuf, utils, workflow
+from trezor.enums import FailureType
+from trezor.messages import Failure
+from trezor.wire.context import UnexpectedMessageException, with_context
+from trezor.wire.errors import ActionCancelled, DataError, Error, UnexpectedMessage
+from trezor.wire.protocol_common import Context, Message
+
+if TYPE_CHECKING:
+    from typing import Any, Callable, Container
+
+    from trezor.wire import Handler, LoadedMessageType, WireInterface
+
+    HandlerFinder = Callable[[Any, Any], Handler | None]
+    Filter = Callable[[int, Handler], Handler]
+
+# If set to False protobuf messages marked with "experimental_message" option are rejected.
+EXPERIMENTAL_ENABLED = False
```

Comment:
Note to self: shouldn't this also be somewhere under `storage/cache` to avoid being wiped by a restart?

Reply:
perhaps, but what we do is re-set it manually from storage anyway, which works even on a freshly started device so :woman_shrugging: